### DISCUSSION TITLE
Ubuntu 22 preview stack: share your feedback

### DISCUSSION THREAD

**Discussion by ofalvai** [bitrise-io/stacks#252] [Ideas]
Hello everyone 👋 

The Ubuntu 22 is [now in public preview](https://bitrise.io/blog/post/major-linux-updates-in-march-2024-test-adapt-and-be-prepared). This is another Linux stack recommended for Android workflows, but based on Ubuntu 22.04 instead of Ubuntu 20.04.

This is a public preview and we want to hear your feedback before making it the default stack for Android projects. If you run into a problem when switching or miss something from the old Ubuntu stack, please add a comment below explaining your setup.

### Important changes

Besides the new OS version, the new stack has a few major changes that we long wanted to make, but we didn't want to break workflows running on the Ubuntu 20 stack. 

**[Summary of changes](https://stacks.bitrise.io/announcements/ubuntu-22/)** is available here.

**Comment by esteluk** [comment]
Hey :wave: A small issue we've found on the new stacks is that `kubectl` doesn't install from `apt` without [manually configuring](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/) the kubernetes apt repository. I'm not completely sure how this worked before, but if the apt repository was part of the 20.04 stack before it would be great to keep it.

**Reply by ofalvai** [reply]
Hey @esteluk, thank you for the report! We looked into this and realized that `kubectl` is actually provided on the Ubuntu 20 stack by the Google Cloud APT repo, which we set up to install the Google Cloud SDK. So we never intended to make `kubectl` available from a repo, it was just a side effect of installing the Google Cloud SDK in a certain way.

On the Ubuntu 22 stack, we switched the installation method of the SDK to the [versioned archives](https://cloud.google.com/sdk/docs/downloads-versioned-archive) way. We'll look into adding the official Kubernetes APT repo to the stack to keep the old behavior.

**Comment by esteluk** [comment]
Sorry me again :wave: It's possibly entirely unrelated, but in the one project we've migrated to 22.04, we've seen significantly increased test flakiness in large Espresso test suites since approximately March 18 when the most recent stack update occurred.

It's the nature of test flakiness that I don't really have a good sense of the root cause yet (and there are only so many experiments on 90 minute test packs I can run in a day!), but since ~12PM CET on the 18th the success rate has dropped to 2/25 executions. I noticed the Stack update and thought it might be related.

**Comment by esteluk** [comment]
Another issue I noticed whilst testing a different project: if I remove the `set-java-version` build step to explicitly use Java 17, the build fails when running `install-missing-android-tools` with:

```
Ensure required Android SDK components
Retrying...
Failed to execute Step:
  failed to install missing android components:
    command failed with exit status 1 (./gradlew "dependencies" "--stacktrace"):
      FAILURE: Build failed with an exception.
      * Where:
      Build file '/bitrise/src/demo/build.gradle' line: 1
      * What went wrong:
      A problem occurred evaluating project ':demo'.
      > Failed to apply plugin 'com.android.internal.application'.
      > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in /usr/lib/jvm/java-11-openjdk-amd64
      You can try some of the following options:
      - changing the IDE settings.
      - changing the JAVA_HOME environment variable.
      - changing `org.gradle.java.home` in `gradle.properties`.
      * Try:
      > Run with --info or --debug option to get more log output.
      > Run with --scan to get full insights.
      > Get more help at https://help.gradle.org/.
```

It works as expected if I still explicitly set Java 17 immediately before this.
